src/components/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,116 @@
+import { Switch } from '@headlessui/react';
+import clsx from 'clsx';
+import React from 'react';
+import styles from './Toggle.module.css';
+
+type ToggleLabelProps = {
+  /**
+   * Visible text label for the toggle.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classnames passed in for styling the toggle label.
+   */
+  className?: string;
+};
+
+type ToggleButtonProps = {
+  /**
+   * Additional classnames passed in for styling the toggle button.
+   */
+  className?: string;
+  /**
+   * Whether toggle is checked or unchecked.
+   */
+  checked: boolean;
+  /**
+   * Callback called when toggle state is changed.
+   */
+  onChange: (checked: boolean) => void;
+  /**
+   * Whether toggle is disabled.
+   */
+  disabled?: boolean;
+};
+
+type ToggleProps = ToggleButtonProps & {
+  /**
+   * When possible, use a visible label through the `label` prop instead.
+   * In rare cases where there's no visible label, you must provide an
+   * `aria-label` for screen readers.
+   */
+  'aria-label'?: string;
+  /**
+   * Visible text label for the toggle.
+   */
+  children?: React.ReactNode;
+  /**
+   * Visible text label for the toggle.
+   */
+  label?: React.ReactNode;
+};
+
+const ToggleLabel = ({ children, className }: ToggleLabelProps) => {
+  const componentClassName = clsx(styles['toggle__label'], className);
+  return <Switch.Label className={componentClassName}>{children}</Switch.Label>;
+};
+
+const ToggleButton = ({ className, checked, ...other }: ToggleButtonProps) => (
+  <Switch
+    checked={checked}
+    className={clsx(
+      styles['toggle__button'],
+      checked
+        ? styles['toggle__button--checked']
+        : styles['toggle__button--unchecked'],
+      className,
+    )}
+    {...other}
+  >
+    {/* This span is the circular part of the toggle */}
+    <span className={styles['toggle__button-circle']} />
+  </Switch>
+);
+
+const ToggleWrapper = Switch.Group;
```

Comment:
In order to use the `<Switch.Label>` headlessui component, it needs to be wrapped in this `Switch.Group` component

Reply:
Maybe add this as a code comment?